node/network/gossip-support/src/lib.rs

```diff
@@ -64,6 +70,11 @@ struct State {
 	// at least a third of authorities the last time.
 	// `None` otherwise.
 	last_failure: Option<Instant>,
+
+	/// First time we did not reach our connectivity treshold.
+	///
+	/// Will be cleared once we reached it.
+	first_failure: Option<Instant>,
```

Comment:
I don't understand the reason for this value. Especially as you reset it below, meaning this isn't always the first failure.

Wouldn't just a `start: Instant` be enough to check that we are above the threshold and start with the printing of the warning?

Reply:
first failure in a sequence of failures. So the first time we did not connect to enough validators of potentially many attempts. This is most important at startup, but I kept it more generic: Whenever we are not connected to at least 2/3rds of all validators for at least a minute, a warning is printed. I'll clarify the comment.